pkg/serverless/invocationlifecycle/invocation_processor.go

```diff
@@ -11,4 +11,6 @@ type InvocationProcessor interface {
 	OnInvokeStart(startDetails *InvocationStartDetails)
 	// OnInvokeEnd is the hook triggered when an invocation has ended
 	OnInvokeEnd(endDetails *InvocationEndDetails)
+	// GetExecutionInfo returns the current execution start information
+	GetExecutionInfo() *ExecutionStartInfo
```

Comment:
I don't think this would help but I'm not entirely sure. We pass in the executionStartInfo into various functions which mutates the values. If this were a copy, the copy values would get mutated instead, and the value inside the `LifecycleProcessor` would not, which means we'd need to pass in a pointer to this value regardless, which seems to defeat the point of using a non-pointer type. For example:

```
executionInfo := ExecutionStartInfo{...} // not a pointer
...
startExecutionSpan(executionInfo, ...) // this function mutates the value of executionInfo, so we'd need to pass in a pointer so that this is reflected in the calling function
```

To me it makes sense to keep this a pointer.